File: BASIC_OF_ALGORITHM/How_to_Solve_N_&_M/NM_3.cpp
//중복가능 수열
#include <iostream>
using namespace std;

int A[10];

void go(int index, int n, int m) {
	if (index == m) {
		for (int i = 0; i < m; i++) {
			cout << A[i] << " ";
		}
		cout << endl;
		return;
	}
	for (int i = 1; i <= n; i++) {
		A[index] = i;
		go(index + 1, n, m);
	}

}


int main() {
	int N, M;
	cin >> N >> M;

	go(0, N, M);










	return 0;
}

File: BASIC_OF_ALGORITHM/How_to_Solve_N_&_M/NM_3.py
A = [0] * 10
def go(index, n, m):
    if index == m:
        for i in range(m):
            print(A[i], end = " ")
        print() 
        return
    
    for i in range(1, n + 1):
        A[index] = i
        go(index + 1, n, m)
N, M = input().split()
go(0, int(N), int (M))

#FOR문의 연산과 비슷하군요


    